### PR TITLE
Add filetype submenus for new file with template menus 

### LINKED ISSUE #3357: Idea: Submenus for the "New" Menu
In Geany's `./template/files/` directory, I think it would be useful to be able to have subdirectories containing files that a user may want to group together and then such files would appear under a submenu of the "New" menu.

The use of this would be for people like me that have numerous templates and especially multiple related templates (like various C or Python templates). One may put their Python files (for example) in a directory called "Python" under `./template/files/` and then the "New" menu would have a "Python" submenu that lists all of the templates in that directory.

### PR DESCRIPTION
**Pull request by ntrel**
Add submenus when there's more than one template per filetype.

Fixes #3357.
Also put menu items with submenus at the top of the menu, not the bottom.

### CODE REVIEW

File: src/templates.c [hunk 1]
```diff
@@ -255,19 +255,57 @@ static void add_file_item(const gchar *fname, GtkWidget *menu)
...
 	foreach_slist(node, list)
 	{
 		gchar *fname = node->data;
+		GeanyFiletype *ft = filetypes_detect_from_extension(fname);
+		FTMenu *group = NULL;
 
-		add_file_item(fname, menu);
+		if (ft)
+			group = &ft_groups[ft->id];
+		if (group->count == 1)
```
Comment (elextr) [inline src/templates.c:291]: `group` may be NULL if a file in the directory does not have a filetype extension

**Review by elextr — CHANGES_REQUESTED** [review]
AFAICT this requires files in the templates directory to have a filetype extension, this is not backward compatible and undocumented as currently files with any name are added to the menu, whereas now Geany would crash (see comment).

There are a lot of file formats that are not supported by a Geany filetype, this change would now preclude the user from having templates for them.

Instead, if a file does not have a filetype extension it should be added to the menu top level like those with a count of 1.

File: src/templates.c (hunk 1 above)
Comment (ntrel) [reply src/templates.c:289]: I realized `ft` is never NULL in this function. The `if (ft)` checks were redundant, removed.

Comment (elextr) [reply src/templates.c:288]: Ahh, ok, `filetypes_detect_from_extension()` returns filetype none if it doesn't recognise it.

**Comment by ntrel** [comment]
OK to merge?

**Comment by elextr** [comment]
Code ok by inspection (I spend blocks of time away from my dev machine and can't build anything, so can't test ATM).

Need to update https://www.geany.org/manual/current/index.html#file-templates to describe using directories.

**Comment by ntrel** [comment]
> Need to update https://www.geany.org/manual/current/index.html#file-templates to describe using directories.

Now mentioned filetype submenus. Directories aren't used.

**Comment by elextr** [comment]
Yeah, thats what I meant ;-)

**Pull request merged**